### DISCUSSION TITLE
Local Avatar Logging

### DISCUSSION THREAD

**Discussion by teknikalarkitekt** [vrcx-team/VRCX#545] [Ideas]
Simply put, I like to get crashers banned. It's super easy. All I have to do is log the avatar, the user ID, and the instance ID and timestamp when somebody tries to crash somebody. After that it's as simple as writing up a detailed support ticket and they're usually banned within a few hours. I used to use a client to log avatars as well as protect me from crashers, but those times are long passed. Now I have to run around with all my safety settings on max, so I can't even see when somebody's using a crash avatar until everybody in the lobby dies. I would like to be able to get these degenerates banned again and I feel like VRCX, with some slight modification, could help with that.
Crashers are super common in the anime communities where I dwell because they "hate weebs".

**Comment by Myrkie** [comment]
VRCX has no access to an avatars data or its id, that is the purpose of the [avatar database](https://avtr.just-h.party/), and that is still only limited to public avatars and that is not a full proof solution since that relies on the avatar database having the avatars information stored. the only way to view an avatars ID would be to read your cache which is a intensive process as well as being against VRChat's terms of service.

**Reply by teknikalarkitekt** [reply]
unfortunate. I was hoping since vrcx lets me know when a friend changes their avatar, I could see when anybody in the lobby does so, and hope to somehow find out which one may have crashed the lobby, at the very least through trial and error. Most crashers use public avatars anyway. and they're usually loud and obnoxious about their actions which makes it easy to tell who it was that changed into the crash avatar in the first place

**Reply by Myrkie** [reply]
> unfortunate. I was hoping since vrcx lets me know when a friend changes their avatar, I could see when anybody in the lobby does so, and hope to somehow find out which one may have crashed the lobby, at the very least through trial and error. Most crashers use public avatars anyway. and they're usually loud and obnoxious about their actions which makes it easy to tell who it was that changed into the crash avatar in the first place

changing avatar logging has nothing to do with ID, the information of the avatar and its name is grabbed from the image since you can request the json for the image which includes its original upload name and author, that's why some avatars can have missing avatar information as the image has been modified in a method that is no longer possible, you would still need a database to view the avatars full information.
